skycatalogs/scripts/rotate.py

```diff
@@ -0,0 +1,250 @@
+        of the point about the new field center
+        """
+        xyz = cartesian_from_spherical(np.radians(ra),
+                                       np.radians(dec)).transpose()
+        xyz = np.dot(self._transformation, xyz).transpose()
+        ra_out, dec_out = spherical_from_cartesian(xyz)
+        return np.degrees(ra_out), np.degrees(dec_out)
+
+
+class GalaxyRotator:
+    def __init__(self, field_rotator, field_name, obj_list):
+        self._field_rotator = field_rotator
+        self._field_name = field_name
+        self._obj_list = obj_list
+        self._stride = 1000000
+
+    def output_field_pixels(self, output_dir, arrow_schema):
+        # First make necessary new columns and keep track of output
+        # healpixels
+        field_distinct = set()
+        colls = self._obj_list.get_collections()
+        new_ra = []
+        new_dec = []
+        new_av_ext = []
+        hps = []
+        hps_distinct = []
+        # Make a dict indexed by hp number.  Value is df to be written
+        # to file for that hp
+        hp_out_dict = dict()
+        for c in colls:
+            ra, dec = self._field_rotator.transform(c._ra, c._dec)
+            new_ra.append(ra)
+            new_dec.append(dec)
+            new_av_ext.append(make_MW_extinction_av(ra, dec))
+            rot_hp = np.array(healpy.pixelfunc.ang2pix(NSIDE, ra, dec,
+                                                       lonlat=True))
+            hps.append(rot_hp)
+            distinct = {h for h in rot_hp}
+            hps_distinct.append(sorted(distinct))
+            field_distinct = set.union(field_distinct, distinct)
+
+        # hp_sorted = sorted(field_distinct)
+        native = colls[0].native_columns
+
+        for i, c in enumerate(colls):
+            to_get = [n for n in native if (n != 'ra') and (n != 'dec') and (n != 'MW_av') and n != 'sed_val_bulge' and n != 'sed_val_disk']
+            dat = c.get_native_attributes(to_get)
+            dat['ra'] = new_ra[i]
+            dat['dec'] = new_dec[i]
+            dat['MW_av'] = new_av_ext[i]
+            dat['hp'] = hps[i]
+            # special handling for for tophat SEDs
+            for k in ['sed_val_bulge', 'sed_val_disk', 'sed_val_knots']:
+                dat[k] = c.get_native_attribute(k, no_np=True)
+
+            df = pd.DataFrame(dat)
+            df.sort_values('hp', inplace=True)
+
+            # Now for hps represented in the collection, copy relevant
+            # rows to df for that hp
+            for hp in hps_distinct[i]:
+                sub_df = (df.loc[df['hp'] == hp]).copy()
+                if hp in hp_out_dict:
+                    hp_out_dict[hp] = pd.concat([hp_out_dict[hp], sub_df])
+                else:
+                    hp_out_dict[hp] = sub_df
+            del df
+
+        # Now write out healpixels for this collection (sets of healpixels
+        # belonging to different fields are disjoint)
+        # for hp in hps_distinct:
```

Comment:
I've removed the last line.   I have my doubts whether the sentence in the first two lines is even true.  It would depend on how close the fields are to each other.